### PR TITLE
Log478 promtail

### PR DESCRIPTION
**Pull request by jcantrill**
Devpreview of promtail deployment https://jira.coreos.com/browse/LOG-478

cc @brancz @sichvoge The last commit is the one of interest and its untested. Care to see if it works?

### CODE REVIEW

**Comment by brancz** [comment]
Discovered 1-2 bugs on my way, but let's fix that as follow ups.

**Comment by brancz** [comment]
Unfortunately I'm getting the same error building locally as CI is seeing.

**Comment by jcantrill** [comment]
@brancz something must of changed with the available environment given everything passed when I initially messaged you.

**Comment by brancz** [comment]
> @brancz something must of changed with the available environment given everything passed when I initially messaged you.

Yes, everything passed when I `/lgtm`ed. There was some sort of CI issues, apparently some are resolved now, let's see if we can still get it to pass.

File: manifests/4.3/collectors.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          description: Specification for log collector
+          properties:
+            type:
+              description: The type of Log Collection to configure
+              type: string
+              enum:
+              - promtail
+            resources:
+              type: object
+              description: The resource requirements for Fluentd
```
Comment (richm) [inline manifests/4.3/collectors.crd.yaml]: s/Fluentd/Promtail/ ?

File: pkg/k8shandler/promtail.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	container.Env = []v1.EnvVar{
+		v1.EnvVar{Name: "PROMTAIL_YAML_HASH", Value: md5hash},
+	}
+	container.Args = []string{
+		"-config.file=/etc/promtail/promtail.yaml",
+		"-client.url=" + collector.PromTailSpec.Endpoint,
+	}
+
+	container.VolumeMounts = []v1.VolumeMount{
+		{Name: "varlog", ReadOnly: true, MountPath: "/var/log"},
```
Comment (richm) [inline pkg/k8shandler/promtail.go]: will need `/run/log` for journald `Volatile` configuration

**Pull request merged**